### PR TITLE
feat(Evaluate): callable objects can be used as metric for Evaluate

### PR DESCRIPTION
**Pull request by lzjever**
Minor changes to enable the evaluator to accept any callable objects as metrics.

### CODE REVIEW

File: dspy/evaluate/evaluate.py [hunk 1]
```diff
@@ -141,8 +142,9 @@ def wrapped_program(example_idx, example):
         # Truncate every cell in the DataFrame
         df = df.applymap(truncate_cell)
 
-        # Rename the 'correct' column to the name of the metric function
-        metric_name = metric.__name__
+        # Rename the 'correct' column to the name of the metric object
+        assert(callable(metric))
```
Comment (insop) [inline dspy/evaluate/evaluate.py:146]: Is the current `metric` already `callable`?

Comment (lzjever) [reply dspy/evaluate/evaluate.py:146]: Getting metric.__name__ only works on function. It fails when the metric is an object with __call__ method

Comment (lzjever) [reply dspy/evaluate/evaluate.py:146]: No. It won't break the existing code, since all current metrics are functions.

**Comment by okhat** [comment]
Nice, this was needed! Thank you @lzjever and thanks for the review @insop !

**Pull request merged**